presto-cache/src/test/java/com/facebook/presto/cache/alluxio/TestAlluxioCachingFileSystem.java

```diff
@@ -0,0 +1,411 @@
+    {
+        assertEquals(MetricsSystem.meter(metricsKey.getName()).getCount() - baseline.getOrDefault(metricsKey.getName(), 0L), expected);
+    }
+
+    private AlluxioCachingFileSystem cachingFileSystem(CacheConfig cacheConfig, AlluxioCacheConfig alluxioCacheConfig)
+            throws URISyntaxException, IOException
+    {
+        Map<Path, byte[]> files = new HashMap<>();
+        files.put(new Path(testFilePath), data);
+        Configuration configuration = getHdfsConfiguration(cacheConfig, alluxioCacheConfig);
+        ExtendedFileSystem testFs = new TestingFileSystem(files, configuration);
+        URI uri = new URI("hdfs://test:8020/");
+        AlluxioCachingFileSystem cachingFs = new AlluxioCachingFileSystem(configuration, uri, testFs, new CacheFactory());
+        cachingFs.initialize(uri, configuration);
+        return cachingFs;
+    }
+
+    private Configuration getHdfsConfiguration(CacheConfig cacheConfig, AlluxioCacheConfig alluxioCacheConfig)
+    {
+        Configuration configuration = new Configuration();
+        if (cacheConfig.isCachingEnabled() && cacheConfig.getCacheType() == ALLUXIO) {
+            configuration.set("alluxio.user.local.cache.enabled", String.valueOf(cacheConfig.isCachingEnabled()));
+            configuration.set("alluxio.user.client.cache.size", cacheConfig.getMaxInMemoryCacheSize().toString());
+            configuration.set("alluxio.user.client.cache.dir", cacheConfig.getBaseDirectory().getPath());
+            configuration.set("alluxio.user.client.cache.size", cacheConfig.getMaxInMemoryCacheSize().toString());
+            configuration.set("alluxio.user.client.cache.page.size", Integer.toString(PAGE_SIZE));
+            configuration.set("alluxio.user.metrics.collection.enabled", String.valueOf(alluxioCacheConfig.isMetricsCollectionEnabled()));
+            configuration.set("alluxio.user.client.cache.async.write.enabled", String.valueOf(alluxioCacheConfig.isAsyncWriteEnabled()));
+            configuration.set("sink.jmx.class", alluxioCacheConfig.getJmxClass());
+            configuration.set("sink.jmx.domain", alluxioCacheConfig.getMetricsDomain());
+        }
+        return configuration;
+    }
+
+    private int readFully(AlluxioCachingFileSystem fileSystem, long position, byte[] buffer, int offset, int length)
+            throws IOException
+    {
+        try (FSDataInputStream stream = fileSystem.open(new Path(testFilePath), buffer.length)) {
+            return stream.read(position, buffer, offset, length);
+        }
+    }
+
+    private static class TestingFileSystem
+            extends ExtendedFileSystem
+    {
+        private final Map<Path, byte[]> files;
+
+        TestingFileSystem(Map<Path, byte[]> files, Configuration conf)
+        {
+            this.files = files;
+            setConf(conf);
+        }
+
+        @Override
+        public FileStatus getFileStatus(Path f)
+                throws IOException
```

Comment:
all `throws IOException`s in this file are redundant

Reply:
Removed.